### PR TITLE
Add basic layout and styling

### PR DESCRIPTION
**Pull request by SageBaram**
* Setup initial project settings
* Background component
* Logic component for questions & answers.
* Basic json file for q&a
* Mui setup, converted HTML to components.
* Main page categories
* Add basic routing to questionary & categories using react-router-dom.
* Configure linters, formatters, ignore files etc.
* Tested branch protection rules - it works!

TODO:
- create a Mui reset theme.
- create context for theme & mui components.

### CODE REVIEW

**Comment by SageBaram** [comment]
Comments didn't go through, added major changes.

File: package.json [hunk 1]
```diff
@@ -29,8 +28,13 @@
     "@typescript-eslint/eslint-plugin": "^6.17.0",
     "@typescript-eslint/parser": "^6.17.0",
     "@vitejs/plugin-react": "^4.2.1",
+    "escape-string-regexp": "^5.0.0",
```
[the remark this answers is not in the document]
Comment (SageBaram) [reply package.json:31]: dependency library

File: src/modules/category/Categories.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  const [selectedCategory, setSelectedCategory] = useState("");
+  const navigate = useNavigate();
+  const { setScore } = useScore();
+
+  useEffect(() => {
+    setScore(0);
+    navigate(selectedCategory.toLowerCase());
+  }, [selectedCategory, setScore, navigate]);
+
+  const handleOnCategoryClick = (category: string) => () => {
```
[the remark this answers is not in the document]
Comment (SageBaram) [reply src/modules/category/Categories.tsx:21]: fixed

File: src/utils/util.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { useParams } from "react-router-dom";
+import allQuestions from "../constants/questions";
+import { QuestionType } from "../constants/types";
+
+export const randomVal = (maxVal: number) => Math.floor((Math.random() * maxVal));
+export const randomIndex = (array: any) => randomVal(array.length);
+export const indexToAlpha = (index: number) => String.fromCharCode(65 + index);
```
[the remark this answers is not in the document]
Comment (SageBaram) [reply src/utils/util.ts:7]: fixed, add a constant asciiLetterConverter const.

File: src/modules/review/Review.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { Box, Typography } from "@mui/material";
+import { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import { useScore } from "../../contexts/ScoreContext";
+
+const Review = () => {
+  const navigate = useNavigate();
+  const { score, setScore } = useScore();
+
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (SageBaram) [reply src/modules/review/Review.tsx:10]: fixed, created a custom hook for timeout

File: src/modules/questionary/Questionary.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    questions[index]
+  );
+
+  const onOptionSelect = (selectedOption: string) => {
+    if (cache.length >= questions.length) {
+      navigate("/review");
+    }
+
+    let index;
+    do {
```
[the remark this answers is not in the document]
Comment (DadonStyle) [reply src/modules/questionary/Questionary.tsx:28]: refactored

File: src/modules/category/Categories.scss [hunk 6]
```diff
@@ -0,0 +1,14 @@
+body .category-container {
+      display: flex;
+      flex-direction: column;
+      justify-content: center;
+      align-items: center;
+      width: 100svw;
+      padding-top: 50%;
+}
+
+body .category-button {
```
[the remark this answers is not in the document]
Comment (SageBaram) [reply src/modules/category/Categories.scss]: fixed

File: src/modules/category/Categories.scss [hunk 7]
```diff
@@ -0,0 +1,14 @@
+body .category-container {
```
[the remark this answers is not in the document]
Comment (SageBaram) [reply src/modules/category/Categories.scss]: fixed

File: src/modules/category/Categories.tsx [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+import { useScore } from "../../contexts/ScoreContext";
+
+const categories: string[] = ["React", "Typescript", "Javascript", "Random"];
+
+const Categories = () => {
+  const [selectedCategory, setSelectedCategory] = useState("");
+  const navigate = useNavigate();
+  const { setScore } = useScore();
+
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (SageBaram) [reply src/modules/category/Categories.tsx:16]: @DadonStyle your todo

File: src/styles/_mixins.scss
[the remark this answers is not in the document]
Comment (SageBaram) [reply src/styles/_mixins.scss:1]: answered below

File: src/modules/review/Review.tsx [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    const timer = setTimeout(() => {
+      navigate("/");
+    }, 4000);
+
+    return () => clearTimeout(timer);
+  }, [setScore, navigate]);
```
[the remark this answers is not in the document]
Comment (SageBaram) [reply src/modules/review/Review.tsx]: not relevant

File: src/styles/_fonts.scss
[the remark this answers is not in the document]
Comment (SageBaram) [reply src/styles/_fonts.scss:1]: answered below

File: src/modules/category/Categories.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (DadonStyle) [reply src/modules/category/Categories.tsx]: resolved

**Review by DadonStyle — APPROVED** [review]

**Pull request merged**